open-craft/xblock-group-project-v2
# Block app config

**Description:** Group Project v2 contents should not be counted towards student course progress.
**JIRA:**
* [MCKIN-3674](https://edx-wiki.atlassian.net/browse/MCKIN-3674)
* [MCKIN-3676](https://edx-wiki.atlassian.net/browse/MCKIN-3676)

**Dependencies:** https://github.com/edx-solutions/edx-platform/pull/586, https://github.com/mckinseyacademy/mcka_apros/pull/1272

**Testing instructions:**
1. Start with fresh course (imported is ok). Course should contain GPv2 section and at least one section with normal course content. Course content should contain at least one HTML XBlock (and, preferably, other blocks that works in Apros and sends progress events).
2. Configure GP v2 for the course.
3. As admin, navigate to participants dashboard: MCKA ADMIN -> Clients -> Click client -> Access Company Dashboard -> Click Course -> Participants
4. Note that all users have zero progress.
5. As one of the users, open Group Project (ideally in other browser/incognito window)
6. Trigger any stage completion (conditions vary; most obvious is "Completion" stage - check the checkbox "I have completed this stage)
7. Note that progress haven't changed in "Participants" page.
8. Open a stage with HTML module in it (i.e. "Upload").
9. Note that progress haven't changed in "Participants" page.
10. Open unit with HTML module in normal course content section
11. Note that progress *have* changed this time
12. If there are any other blocks in the course - test that triggering progress events on them does update progress in "Participants" page
